sdks/java/core/src/main/java/org/apache/beam/sdk/util/Histogram.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util;
+
+import java.math.RoundingMode;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.math.DoubleMath;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** A histogram that supports estimated percentile with linear interpolation. */
+public class Histogram {
```

Comment:
Personally, I think this Histogram class is sufficient for now, and we should proceed with this for now.
There may be some arguments to use a histogram library in the future. I.e. to support sparsely populated histograms.
Please add a TODO to consider implementing with other potential options, such as.

Apache Commons
https://www.baeldung.com/apache-commons-frequency

GitHub Hdr Histogram
https://github.com/HdrHistogram/HdrHistogram

Reply:
Done.

Reply:
Please add backwards compatibility warnings to this so that we can remove this without breaking users.
Also, could you clarify why we could not use an existing implementation ?

Reply:
I couldn't find any 3rd party library that satisfies our interface requirements such as exponential or explicit bucket definitions. Also, the current usecase is simple enough that we could avoid introducing additional dependencies.